mm2src/lp_ordermatch.rs

```diff
@@ -3131,6 +3145,115 @@ pub async fn set_price(ctx: MmArc, req: Json) -> Result<Response<Vec<u8>>, Strin
     Ok(try_s!(Response::builder().body(res)))
 }
 
+//update makerorder will update the previously created order with new volume and new price.
+pub async fn update_maker_order(ctx: MmArc, req: Json) -> Result<Response<Vec<u8>>, String> {
+    let req: MakerOrderUpdateReq = try_s!(json::from_value(req));
+
+    let ordermatch_ctx = try_s!(OrdermatchContext::from_ctx(&ctx));
+    let mut my_orders = ordermatch_ctx.my_maker_orders.lock().await;
+
+    //Get the previous order details using uuid
+    let my_previous_order = match my_orders.get(&req.uuid) {
```

Comment:
Please consider using `get_mut` and change only required fields based on their existence in the request.